src/util/state_lib/state_publisher_server.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import annotations
+
+import threading
+import time
+from typing import Callable
+
+import rospy
+from mrover.msg import StateMachineStructure, StateMachineTransition, StateMachineStateUpdate
+
+from util.state_lib.state_machine import StateMachine
+
+
+class StatePublisher:
+    structure_publisher: rospy.Publisher
+    state_publisher: rospy.Publisher
+    state_machine: StateMachine
+    __struct_thread: threading.Thread
```

Comment:
This is still named _struct_thread even though you use self._structure_thread below